tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -425,7 +446,13 @@ class FileInfo:
         Tuple[NodeAndRange, chapel.TypedSignature]
     ] = field(init=False)
     uses_here: Dict[str, References] = field(init=False)
-    instantiations: Dict[str, Set[chapel.TypedSignature]] = field(init=False)
+    instantiations: Dict[
+        str,
+        Dict[
+            chapel.TypedSignature,
+            List[Tuple[chapel.FnCall, Optional[chapel.TypedSignature]]],
```

Comment:
this is now very long and confusing. Can you extract pieces of it into a type alias?